keras_nlp/models/llama3/llama3_preprocessor.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.llama3.llama3_tokenizer import Llama3Tokenizer
+from keras_nlp.models.llama.llama_preprocessor import LlamaPreprocessor
+
+
+@keras_nlp_export("keras_nlp.models.Llama3Preprocessor")
+class Llama3Preprocessor(LlamaPreprocessor):
+    tokenizer_cls = Llama3Tokenizer
+
+    def __init__(
+        self,
+        tokenizer,
+        sequence_length=1024,
+        add_start_token=False,
+        add_end_token=False,
+        **kwargs
+    ):
+        super().__init__(
+            tokenizer, sequence_length, add_start_token, add_end_token, **kwargs
```

Comment:
Just want to make sure I understand correctly: is the main difference between llama3 preprocessor and llama preprocessor that `add_start_token=False` in llama3 and `add_start_token=True` in llama?
Or are there other differences?

Reply:
Yes, as far as the preprocessor is concerned, this is the only change. But there's one more change in the tokenizer, the authors don't provide a sentence piece tokenizer anymore. Instead, they provide a BPE vocab and merges file, so I had to create a completely new preprocessing pipeline for LLaMA 3.

Reply:
I see!
re new directories for llama3, what do you think about adding new llama3 classes to llama files?
* llama
   *  llama_tokenizer.py
       * LlamaTokenizer
       * Llama3Tokenizer
   * llama_preprocessor.py
       * LlamaPreprocessor
       * Llama3Preprocessor
 * etc

This way, we won't have a new llama3 directory and files.

Reply:
Good idea. I went for a different directory to maintain the precedence in KerasNLP. I will check if it's possible to do this without making the existing files too cumbersome.

What do you think about https://github.com/keras-team/keras-nlp/pull/1584#pullrequestreview-2012132854? I am curious if there's a better way to utilize the current infra to avoid creating separate classes. One way I can think of is to accept both `proto` and `vocab, merges` in the tokenizer class and branch to BPE or sentence piece based on the inputs. I avoided that since it an unclean approach and also pollutes the API a bit.